addons/website/static/src/js/editor/snippets.options.js

```diff
@@ -4091,8 +4091,15 @@ options.registry.GalleryElement = options.Class.extend({
 });
 
 options.registry.Button = options.Class.extend({
-    forceDuplicateButton: true,
-
+    /**
+     * @override
+     */
+    init() {
+        this._super(...arguments);
+        const isCtaButton = this.$target[0].matches?.(".navbar a.btn_cta.oe_unremovable");
```

Comment:
Indeed, it's not needed here. It's just the habit of securing the `.matches()` calls when the target is not an instance of `Element` also It seems more correct to prevent the duplication of "unremovable" buttons as you suggested.
Diff updated now, thanks :+1:

Reply:
Hmm... Actually, the short-circuiting condition in optional chaining is always based on checking if the "left operand" is `undefined` or `null`, so it should be: `target.matches?.("...")` for calling a method that may not exist, but I think it's something to be careful with in general: E.g. if the target has a `matches` "attribute", the code will still trigger an error since  `matches` is not a function :grimacing: (see: [details here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining#optional_chaining_with_function_calls)).